core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackend.scala

```diff
@@ -273,25 +277,25 @@ private[spark] class CoarseMesosSchedulerBackend(
         matchesAttributeRequirements(slaveOfferConstraints, offerAttributes)
       }
 
-      declineUnmatchedOffers(d, unmatchedOffers)
+      unmatchedOffers.foreach { offer =>
+        declineOffer(d, offer, "unmet constraints", rejectOfferDurationForUnmetConstraints)
+      }
+
       handleMatchedOffers(d, matchedOffers)
     }
   }
 
-  private def declineUnmatchedOffers(d: SchedulerDriver, offers: Buffer[Offer]): Unit = {
-    for (offer <- offers) {
-      val id = offer.getId.getValue
-      val offerAttributes = toAttributeMap(offer.getAttributesList)
-      val mem = getResource(offer.getResourcesList, "mem")
-      val cpus = getResource(offer.getResourcesList, "cpus")
-      val filters = Filters.newBuilder()
-        .setRefuseSeconds(rejectOfferDurationForUnmetConstraints).build()
+  private def declineOffer(d: SchedulerDriver, offer: Offer, reason: String, refuseSeconds: Long) {
+    val id = offer.getId.getValue
+    val offerAttributes = toAttributeMap(offer.getAttributesList)
+    val mem = getResource(offer.getResourcesList, "mem")
+    val cpus = getResource(offer.getResourcesList, "cpus")
 
-      logDebug(s"Declining offer: $id with attributes: $offerAttributes mem: $mem cpu: $cpus"
-        + s" for $rejectOfferDurationForUnmetConstraints seconds")
+    logDebug(s"Declining offer ($reason): $id with attributes: $offerAttributes mem: $mem"
```

Comment:
Place the "Reason: " field somewhere more readable.  Right now, it breaks up the log message at a confusing location.